src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -171,17 +171,20 @@ protected void processFiltered(File file, List<String> lines) throws CheckstyleE
             final String msg = "%s occurred during the analysis of file %s.";
             final String fileName = file.getPath();
             try {
-                final FileText text = FileText.fromLines(file, lines);
-                final FileContents contents = new FileContents(text);
-                final DetailAST rootAST = parse(contents);
-
-                getMessageCollector().reset();
-
-                walk(rootAST, contents, AstState.ORDINARY);
-
-                final DetailAST astWithComments = appendHiddenCommentNodes(rootAST);
-
-                walk(astWithComments, contents, AstState.WITH_COMMENTS);
+                if (!ordinaryChecks.isEmpty()
+                        || !commentChecks.isEmpty()) {
+                    final FileText text = FileText.fromLines(file, lines);
+                    final FileContents contents = new FileContents(text);
+                    final DetailAST rootAST = parse(contents);
+                    getMessageCollector().reset();
```

Comment:
Can we keep original line separation between statements?
So empty line before and after here.